src/main/java/seedu/toluist/model/Task.java

```diff
@@ -61,4 +61,22 @@ public void replaceTags(ArrayList<Tag> tags) {
     public boolean isOverdue() {
         return startDateTime != null && startDateTime.isBefore(LocalDateTime.now());
     }
+
+    public boolean isStringContainedInDescriptionIgnoreCase(String comparison) {
+        return this.description.toLowerCase().contains(comparison.toLowerCase());
+    }
+
+    public boolean isStringContainedInAnyTagIgnoreCase(String comparison) {
+        for (Tag tag : allTags) {
+            if (tag.tagName.toLowerCase().contains(comparison.toLowerCase())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public int compareTo(Task comparison) {
```

Comment:
Fair point, but only important if we allow different sorting options for the user. Otherwise we can still implement our sorting method here, accounting for that.